src/api/images.tsx

```diff
@@ -114,3 +115,17 @@ export const createImageAlias = (
       .catch(reject);
   });
 };
+
+export const exportImage = (
+  fingerprint: string,
+  onDownload: (response: Response) => Promise<void>,
+): Promise<void> => {
+  return new Promise((resolve, reject) => {
```

Comment:
Please could you explain your reasoning?

Reply:
I mean `fetch` itself returns a promise. Could this part be simplified to
```
export const exportImage = (
  fingerprint: string,
  onDownload: (response: Response) => Promise<void>,
): Promise<Response> => fetch(`/1.0/images/${fingerprint}/export`).then(handleRawResponse).then(onDownload);
```
...for example.

Or
```
export const exportImage = (fingerprint: string): Promise<Response> => fetch(`/1.0/images/${fingerprint}/export`).then(handleRawResponse);
```
I think, there is no need to pass `onDownload` handler from the upper level and then return the result back there. It can be done from the component where it was called.

I haven't tested it though. Types might cause errors.

Eventually, what's left here is the hardcoded URL and that `handleRawResponse` function that transforms the raw response into a more convenient format. This can be solved with `axios` (already discussed that with @mas-who), so this  `exportImage` function might not be needed at all.

Reply:
Thanks @yurii-vasyliev yes agreed that we don't have to pass `onDownload` here. I think for `axios`, we still need to decide if we will do a project wide refactor for it, so might be good to keep that change out of the context of this PR. 

@edlerd we should have a discussion about whether if we want to use `axios` for lxd-ui. I think there might be value in adopting the library to align the approach with other projects, wdyt?

Reply:
@mas-who , sure, I didn't mean to include `axios` to this PR.

Reply:
Thank you for your explanation, @yurii-vasyliev